File: 프로그래머스/lv1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    answer = ''
    x_list = [0] * 10
    y_list = [0] * 10
    
    for i in X:
        x_list[int(i)] += 1
    
    for i in Y:
        y_list[int(i)] += 1
        
    for i in range(9, -1, -1):
        if x_list[i] >= 1 and y_list[i] >= 1:
            cnt = min(x_list[i], y_list[i])
            answer += str(i) * cnt
    if len(answer) == 0: return "-1"
    elif answer[0] == "0": return "0"
    

    return answer

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.js
function solution(X, Y) {
    var answer = '';
    const x = new Array(10).fill(0)
    const y = new Array(10).fill(0)

    for (let i = 0; i < X.length; i++) {
        const index = Number(X[i])
        x[index] += 1
    }
    
    for (let i = 0; i < Y.length; i++) {
        const index = Number(Y[i])
        y[index] += 1
    }
    
    for (let i = 0; i < 10; i++) {
        const xCount = x[i]
        const yCount = y[i]
        if (xCount >= 1 && yCount >= 1) {
            answer = i.toString().repeat(Math.min(xCount, yCount)) + answer
        }
    }
    
    if (answer.length === 0) answer = "-1"
    else if (Number(answer) === 0) answer = "0"
    
    return answer;
}